nixops/script_defs.py

```diff
@@ -236,20 +238,20 @@ def name_to_key(name: str) -> Tuple[str, str, List[object]]:
         for name in names:
             d = definitions.get(name)
             r = depl.resources.get(name)
-            assert r is not None
-            if deployment.is_machine(r):
+
+            resource_state: str = "Missing"
+            if r is not None and deployment.is_machine(r):
```

Comment:
The logic here and in `state()` was freaking me out, so I traced it out a bit and tried to simplify it some: https://github.com/adisbladis/nixops/pull/1 I'm not ... 100% ... confident ... that it is the same thing. Take a look?

Reply:
The logic in `state` is not exactly the same, instead of implicitly returning `None` i raise an error.
The old behaviour in that function would result in a string containing `None` (as a string).

Reply:
This block behaves the same as before adding the assertion.